src/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -54,16 +53,44 @@ public ICslQueryProvider GetProvider()
             if (value != null)
                 return value;
 
-            _kustoQueryProvider = value = KustoClientFactory.CreateCslQueryProvider(_options.CurrentValue.QueryConnectionString);
+            _kustoQueryProvider = value = KustoClientFactory.CreateCslQueryProvider(GetKustoConnectionStringBuilder());
             return value;
         }
     }
 
     private string DatabaseName => _options.CurrentValue.Database;
+    private string KustoClusterUri => _options.CurrentValue.KustoClusterUri;
+    private string ManagedIdentityId => _options.CurrentValue.ManagedIdentityId;
+    private bool UseAzCliAuthentication => _options.CurrentValue.UseAzCliAuthentication;
+
+    private KustoConnectionStringBuilder GetKustoConnectionStringBuilder()
+    {
+        if (string.IsNullOrEmpty(KustoClusterUri))
+        {
+            throw new ArgumentException($"{nameof(KustoOptions.KustoClusterUri)} is not configured in app settings");
+        }
+        if (string.IsNullOrEmpty(DatabaseName))
+        {
+            throw new ArgumentException($"{nameof(KustoOptions.Database)} is not configured in app settings");
+        }
```

Comment:
why is this enforced here❔ `DatabaseName` isn't used in this method. I agree it should be checked but this location is suspect. Better done in both `ExecuteKustoQueryAsync(...)` and `ExecuteStreamableKustoQuery(...)`

Reply:
I put it there to avoid code duplication, but I agree it doesn't make a lot of sense. I think you're right, better to check right before it's usage